substrate/coins/pallet/src/lib.rs

```diff
@@ -142,15 +157,21 @@ pub mod pallet {
     ///
     /// Errors if any amount overflows.
     pub fn mint(to: Public, balance: Balance) -> Result<(), Error<T, I>> {
-      // update the balance
-      Self::increase_balance_internal(to, balance)?;
-
-      // update the supply
       let new_supply = Self::supply(balance.coin)
         .checked_add(balance.amount.0)
         .ok_or(Error::<T, I>::AmountOverflowed)?;
+
+      // skip the economics check for lp tokens.
+      if TypeId::of::<I>() != TypeId::of::<LiquidityTokensInstance>() &&
```

Comment:
Shouldn't we remove this entirely and just have the LiqTokensInstance use () for AllowMint?